### ISSUE TITLE
[Bug] [iOS] Expander + CollectionView: expanding wrong views.

### ISSUE THREAD

**Issue by AndreiMisiukevich** [xamarin/Xamarin.Forms#10288]
### Description
Expanding several other cells during expanding any "cell".
If then you try to scroll collection a little, the bug will go.

I assume there is a bug somewhere in TemplatedCell of iOS-specific code.

**iOS** Only.
### Steps to Reproduce
1) Download sample
2) Run it on iOS simulator
3) Try to expand any cell

**Works fine on Android.**

### Expected Behavior
Expanding only one "cell" like on Android that was pressed.

### Actual Behavior
Expanding several "cells".

### Basic Information

- Version with issue:
- Last known good version: none
- IDE: VS for Mac
- Platform Target Frameworks: 
  - iOS:  latest
- Affected Devices: iPhone 11 Simulator

### Screenshots
![Apr-08-2020 00-25-41](https://user-images.githubusercontent.com/10124814/78721073-88934c80-792f-11ea-8cc0-31d33cbb0749.gif)

### Reproduction Link
https://github.com/xamarin/Xamarin.Forms/files/4440579/ExpanderTest.zip

**Comment by ericlira** [comment]
I'm having a similar problem but with Android. Openning a Expander inside a CollectionView causes every 13th item to be expanded too.

![expander_issue](https://user-images.githubusercontent.com/22410658/84115224-0b8c6c80-aa71-11ea-963b-0208aa5d6b5c.gif)

**Comment by AndreiMisiukevich** [comment]
> I'm having a similar problem but with Android. Openning a Expander inside a CollectionView causes every 13th item to be expanded too.
> 
> ![expander_issue](https://user-images.githubusercontent.com/22410658/84115224-0b8c6c80-aa71-11ea-963b-0208aa5d6b5c.gif)

Hi) That's certainly a different problem)
Can you replicate this problem on the latest pre-release package?

Please fill a separate ticket

**Comment by ericlira** [comment]
> > I'm having a similar problem but with Android. Openning a Expander inside a CollectionView causes every 13th item to be expanded too.
> > ![expander_issue](https://user-images.githubusercontent.com/22410658/84115224-0b8c6c80-aa71-11ea-963b-0208aa5d6b5c.gif)
> 
> Hi) That's certainly a different problem)
> Can you replicate this problem on the latest pre-release package?
> 
> Please fill a separate ticket

Yeah I just did some more tests and the problem is with CollectionView itself, not Expander, most likely related to CachingStrategy as the same problem occours with Syncfusion expander but not if I replace the CollectionView for ListView with default CachingStrategy.

I'll open another ticket!

**Comment by marksarcade** [comment]
Having the same issue here on iOS. Interestingly it's all find in Android. If it's relevant too, I have an expander outside / above my collection view as a page header, on Android it expends/closes smoothly but on iOS it stutters open/closed.

**Comment by amiraelsayedbadwy** [comment]
have the same issue in ios

**Comment by jchaves07** [comment]
You can try with the AdvanceExpanderPage,  using Stacklayout with datasource instead of ListView, in my case works fine with Stacklayout https://github.com/xamarin/xamarin-forms-samples/tree/master/UserInterface/ExpanderDemos
<img width="933" alt="image" src="https://user-images.githubusercontent.com/10256328/90552187-ac716500-e14f-11ea-9058-952b3c9b4d83.png">

**Comment by rventuri76** [comment]
> You can try with the AdvanceExpanderPage, using Stacklayout with datasource instead of ListView, in my case works fine with Stacklayout https://github.com/xamarin/xamarin-forms-samples/tree/master/UserInterface/ExpanderDemos
> <img alt="image" width="933" src="https://user-images.githubusercontent.com/10256328/90552187-ac716500-e14f-11ea-9058-952b3c9b4d83.png">

Not working for me (4.8.0.126)
